server/src/main/java/org/candlepin/service/model/ProductInfo.java

```diff
@@ -126,4 +128,21 @@
      *  the last update date for this product, or null if the last update date has not been set
      */
     Date getUpdated();
+
+    /**
+     * Utility method that returns a string comparator for comparing productIDs.
+     *
+     * @return A comparator for comparing productIds.
+     */
+    static Comparator<String> getProductIdComparator() {
+        return (lhs, rhs) -> {
+            if (lhs != null && rhs != null) {
```

Comment:
Either use the EqualsBuilder or finish the manual comparison here. Doing the null checks in this way is a bit redundant, and in this particular case, gives us the wrong result when both are null.

Given that this is used in only one place and it's testing an immutable class, this may not be something that requires a factory method. Would probably be cleaner as a one-off closure as needed.

Reply:
Decided to remove comparator, since we have stream of productID's that need equality check. Utility method work just fine for this case.